superduperdb/base/document.py

```diff
@@ -357,12 +358,19 @@ def _deep_flat_decode(r, cache, blobs, files={}, db: t.Optional['Datalayer'] = N
 
 
 def _get_from_reference(r, cache, blobs, files={}, db: t.Optional['Datalayer'] = None):
-    if r[1:] in cache:
-        return _get_leaf_from_cache(r[1:], cache, blobs, files, db=db)
+    if r.startswith('&:blob:'):
+        file_id = r.split('&:blob:')[-1]
+        blob_data = blobs.get(file_id)
```

Comment:
If the data has not been inserted into the database and the referenced data is stored in blobs, check first if the data already exists; if not, retrieve it from the artifact store.